src/profiler.cpp

```diff
@@ -519,11 +520,49 @@ void Profiler::updateThreadName(jvmtiEnv* jvmti, JNIEnv* jni, jthread thread) {
         jvmtiThreadInfo thread_info;
         if (vm_thread != NULL && jvmti->GetThreadInfo(thread, &thread_info) == 0) {
             setThreadName(vm_thread->osThreadId(), thread_info.name);
+            addThreadToFilteredList(vm_thread->osThreadId(), thread_info.name);
             jvmti->Deallocate((unsigned char*)thread_info.name);
         }
     }
 }
 
+void Profiler::addThreadToFilteredList(int osThreadId, const char *name) {
+    MutexLocker lock(_filtered_tids_mutex);
+
+    if (_filter_threads == NULL) return;
+
+    if (strstr(name, _filter_threads)) {
+        _filtered_tids.push_back(osThreadId);
```

Comment:
I'd say `vector` is not a proper structure for holding filtered tids.
One thread id may appear in `vector` multiple times, and the `vector` may grow infinitely.